### PR TITLE
Ticket 6

### PR DESCRIPTION
**Pull request by dan77uk**

### CODE REVIEW

**Review by Harrison365 — CHANGES_REQUESTED** [review]
Hey Dan :) 
Please...
-Test that the returning comments have the correct article_id key.
-Test for an article which exists but that has no comments (this can't be done using .length alone).

Let us know when you want us to take another look. Progress is good so far :)

File: __tests__/getArticleComments.test.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  });
+
+  it("should if article_id has no comments", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(404)
+      .then(({ body }) => {
+        expect(body.msg).toBe("Not found");
+      });
+  });
```
Comment (emilyb93) [inline __tests__/getArticleComments.test.js]: I would disagree with this one in that, if an article does exist but has no comments, we should send back a 200 with an empty array. This is to ensure we dont cause bugs on the front end with data type inconsistency

File: models/selectArticleComments.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+const db = require("../db/connection");
+const { checkArticleExists } = require("../utils/checkArticleExists");
+
+exports.selectArticleComments = (article_id) => {
+  if (/[A-Za-z]/g.test(article_id)) {
+    return Promise.reject({
+      status: 400,
+      msg: "Invalid article id",
+    });
+  }
```
Comment (emilyb93) [inline models/selectArticleComments.js:10]: This logic can be done within SQL as SQL will throw an error if you attempt to use the wrong data type in a query so would prefer that this is done by using an error handle to check the error code

**Review by emilyb93 — CHANGES_REQUESTED** [review]
Change your test but the other one is optional

**Review by hevdevs — APPROVED** [review]
Looks good to merge based on suggestions! 🎉✅

**Pull request merged**